nixos/modules/security/pam.nix

```diff
@@ -253,6 +267,8 @@ let
               "auth sufficient ${pkgs.pam_u2f}/lib/security/pam_u2f.so"}
           ${optionalString cfg.usbAuth
               "auth sufficient ${pkgs.pam_usb}/lib/security/pam_usb.so"}
+          ${optionalString cfg.googleAuth
+              "auth required ${pkgs.googleAuthenticator}/lib/security/pam_google_authenticator.so"}
```

Comment:
I didn't test that case, and I agree; I don't see how to support it without a major refactor. What I've tested is:

- Not using Google Authenticator. The code doesn't break anything if it isn't enabled.
- Logging in over ssh with GA enabled, but with a public key. It doesn't ask for an OTP, as intended.
- Logging in over ssh with GA enabled, and no public key. In this case it asks for OTP *first* (which is important, as otherwise it might be possible to tell whether you've found the right password), then goes on to ask for password. Assuming password authentication is enabled, of course. If it isn't, then it asks for OTP and goes on to fail.

Not tested:
- Logging in with OTP and some other authentication mechanism, because I don't have any.

It isn't ideal, but I think this is still an improvement on the current state.